### PR TITLE
Fix output folder for absolute path inputs

### PR DESCRIPTION
**Pull request by davispuh**
Currently if you use absolute path for input files then output folder is not respected.

That is, this fails:
```
$ dstep -o /tmp /usr/include/stdio.h /usr/include/stdlib.h 
dstep: an unknown error occurred: std.file.FileException@std/file.d(840): /usr/include/stdio.d: Permission denied
----------------
/usr/include/dlang/dmd/std/file.d:283 @trusted bool std.file.cenforce!(bool).cenforce(bool, scope const(char)[], scope const(char)*, immutable(char)[], ulong) [0x562fb05e0003]
??:? @trusted void std.file.writeImpl(scope const(char)[], scope const(char)*, scope const(void)[], bool) [0x562fb06dae0b]
/usr/include/dlang/dmd/std/file.d:745 @safe void std.file.write!(immutable(char)[]).write(immutable(char)[], const(void[])) [0x562fb0639917]
dstep/translator/Translator.d:75 void dstep.translator.Translator.Translator.translate() [0x562fb06b35e0]
dstep/driver/Application.d:66 void dstep.driver.Application.Application.run() [0x562fb065a15c]
dstep/driver/CommandLine.d:181 int dstep.driver.CommandLine.run(immutable(char)[][]) [0x562fb066aa61]
dstep/main.d:18 _Dmain [0x562fb066d977]
std.file.FileException@std/file.d(840): /usr/include/stdio.d: Permission denied
----------------
/usr/include/dlang/dmd/std/file.d:283 @trusted bool std.file.cenforce!(bool).cenforce(bool, scope const(char)[], scope const(char)*, immutable(char)[], ulong) [0x562fb05e0003]
??:? @trusted void std.file.writeImpl(scope const(char)[], scope const(char)*, scope const(void)[], bool) [0x562fb06dae0b]
/usr/include/dlang/dmd/std/file.d:745 @safe void std.file.write!(immutable(char)[]).write(immutable(char)[], const(void[])) [0x562fb0639917]
dstep/translator/Translator.d:75 void dstep.translator.Translator.Translator.translate() [0x562fb06b35e0]
dstep/driver/Application.d:66 void dstep.driver.Application.Application.run() [0x562fb065a15c]
dstep/driver/CommandLine.d:181 int dstep.driver.CommandLine.run(immutable(char)[][]) [0x562fb066aa61]
dstep/main.d:18 _Dmain [0x562fb066d977]
```

It tries to create `.d` files in same directory as input file without respecting output folder.
This PR fixes that so that it will correctly use output folder for such case.

### CODE REVIEW

File: dstep/driver/Util.d [hunk 1]
```diff
@@ -15,3 +15,32 @@ string makeDefaultOutputFile(string inputFile, bool useBaseName = true)
...
+    foreach (path; absolutePaths[1..$])
+    {
+        prefix = commonPrefix(prefix, path);
+    }
+    auto pos = lastIndexOf(prefix, '/');
+    if (pos > 0)
+    {
+        return prefix[0..pos];
+    }
+    return "/";
```
[the remark this answers is not in the document]
Comment (davispuh) [reply dstep/driver/Util.d]: That would be great if there actually were functions like `commonPath`, `rootPath` but it's hallucination. They don't exist and I don't have access to Windows system to test how it should be implemented there to support this case.

[the remark this answers is not in the document]
Comment (jacob-carlborg) [reply dstep/driver/Util.d]: I think Unix style paths work on Windows too. None of the functions used in `findBasePath` touches the file system. Perhaps you can cross-compile? findBasePath should work at CTFE so can test that in isolation without having to run it on Windows.

**Review by jacob-carlborg — CHANGES_REQUESTED** [review]
There are no new tests added that shows the change is working.

File: dstep/driver/Util.d (hunk 1 above)
[the remark this answers is not in the document]
Comment (davispuh) [reply dstep/driver/Util.d]: Yeah `/` works in Windows but there are edge cases with mixed slashes aswell like `D:/a\b/d.h` and drive letters.

**Comment by davispuh** [comment]
I rewrote this PR and added tests. Implement support for Windows aswell and tested that it works correctly.
